stl/inc/regex

```diff
@@ -280,7 +280,8 @@ public:
 
     template <class _FwdIt>
     string_type transform(_FwdIt _First, _FwdIt _Last) const { // apply locale-specific transformation
-        return _Getcoll()->transform(_First, _Last);
+        string_type _Str(_First, _Last);

```

Comment:
Peta-nitpick: In C++14-mode, `_Str.data()` returns a pointer to `const`, but in C++17-and-later a pointer-to-mutable. Formally, if someone links together a C++14 TU and a C++17 TU that instantiate the same specialization of this function template we're violating the One Definition Rule. I can't imagine this ODR violation causing actual problems, but it's easy enough to fix by making `_Str` a constant so the next line calls the same `data` in both C++14 and C++17 modes.
```suggestion
        const string_type _Str(_First, _Last);
```

Reply:
This is so ridiculously esoteric that I feel silly requesting changes, so I'm going to instead approve, apply the change myself, and hope nobody ever reads this ;)

Reply:
@StephanTLavavej, I should call your attention to the fact that I made this silly change after you approved.